tests/phpunit/api/v3/PaymentTest.php

```diff
@@ -406,4 +406,61 @@ public function testUpdatePayment() {
     ));
   }
 
+  /**
+   * Test create payment api for paylater contribution
+   */
+  public function testCreatePaymentPayLater() {
+    $contributionParams = array(
+      'total_amount' => 100,
+      'currency' => 'USD',
+      'contact_id' => $this->_individualId,
+      'financial_type_id' => 1,
+      'contribution_status_id' => 2,
+      'is_pay_later' => 1,
+    );
+    $contribution = $this->callAPISuccess('Contribution', 'create', $contributionParams);
+
+    //Create partial payment
+    $params = array(
+      'contribution_id' => $contribution['id'],
+      'total_amount' => 100,
+    );
+    $payment = $this->callAPIAndDocument('Payment', 'create', $params, __FUNCTION__, __FILE__);
+    $expectedResult = array(
+      'from_financial_account_id' => 7,
+      'to_financial_account_id' => 6,
+      'total_amount' => 100,
+      'status_id' => 1,
+      'is_payment' => 1,
+    );
+    $this->checkPaymentResult($payment, $expectedResult);
+
+    // Check entity financial trxn created properly
+    $params = array(
+      'entity_id' => $contribution['id'],
+      'entity_table' => 'civicrm_contribution',
+      'financial_trxn_id' => $payment['id'],
+    );
+
+    $eft = $this->callAPISuccess('EntityFinancialTrxn', 'get', $params);
```

Comment:
You might also find it helpful to use the test helper function callAPISuccessGetValue - which also lets you pass in the value you are checking against